ts/test/test-helpers.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Makes deep copy of the object o.
+ */
+export function copy<T>(o: T): T {
```

Comment:
Deep cloning an object in JavaScript is tricky. Just use [a module from npm](https://stackoverflow.com/questions/4459928/how-to-deep-clone-in-javascript), instead of writing your own copy.

Reply:
Done.